### PR TITLE
Find a user’s most recently updated repositories in their profile

### LINKED ISSUE #1423: Show recently modified repos on profile landing pages
Maybe the 10 most recently modified repos? Where would be a good space for it?

---

*For anyone that wants this, please also send this feature request to GitHub: support@github.com*

### PR DESCRIPTION
**Pull request by connorbrathwaite**
#1423 Seemed straightforward enough. Will update readme after review!

screen shot 2019-02-19 at 23 30 34

Fixes #1423

### CODE REVIEW

**Comment by busches** [comment]
Looks like there's extra spacing between "Top repositories" and the next /

**Comment by connorbrathwaite** [comment]
screen shot 2019-02-14 at 6 31 59 pm

whoops, sorted.

File: source/features/show-user-updated-repositories.tsx [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+import features from '../libs/features';
+import {getCleanPathname} from '../libs/utils';
+
+function init() {
+	const showcaseTitle = select('.js-pinned-repos-reorder-container .text-normal');
+	if (!showcaseTitle) {
+		return false;
+	}
+
+	const url = `/search?o=desc&q=user%3A${getCleanPathname()}&s=updated&type=Repositories`;
```
Comment (sindresorhus) [inline source/features/show-user-updated-repositories.tsx]: This URL should be properly constructed with `URL` and `URLSearchParams` for readability, ease of future changes, and possible escaping.

**Comment by connorbrathwaite** [comment]
> I think it’s best to merge the two features, so you can just add more parameters to `firstChild.after` and avoid hiding the `reorder-message` in the css
> 
> [/source/features/show-user-top-repositories.tsx@`master`](https://github.com/sindresorhus/refined-github/blob/master/source/features/show-user-top-repositories.tsx)

anything else i could contribute to the pr before documenting the readme?

File: source/features/show-user-top-repositories.tsx [hunk 2]
```diff
@@ -8,14 +8,29 @@ import select from 'select-dom';
 import features from '../libs/features';
 import {getCleanPathname} from '../libs/utils';
 
+const divider = ' / ';
+
+function buildUrl(queryField) {
+	const url = new URL('/search', window.location.href);
```
Comment (bfred-it) [inline source/features/show-user-top-repositories.tsx]:
```suggestion
	const url = new URL('/search', location.href);
```

File: source/features/show-user-top-repositories.tsx [hunk 3]
```diff
@@ -8,14 +8,29 @@ import select from 'select-dom';
...
 	if (!showcaseTitle) {
 		return false;
 	}
 
-	const url = `/search?o=desc&q=user%3A${getCleanPathname()}&s=stars&type=Repositories`;
-	showcaseTitle.firstChild.after(' / ', <a href={url}>Top repositories</a>);
+	showcaseTitle.firstChild.after(
+		divider,
+		<a href={buildUrl('stars')}>Top repositories</a>,
+		divider,
```
Comment (bfred-it) [inline source/features/show-user-top-repositories.tsx]: `' / '` should probably stay inlined, it's only used twice.

**Comment by bfred-it** [comment]
If you tested it and it works, after these two changes and the readme update it can be merged.

File: readme.md [hunk 4]
```diff
@@ -137,6 +137,7 @@ GitHub Enterprise is also supported. More info in the options.
 - [Badge on user's profile if they follow you.](https://user-images.githubusercontent.com/3723666/45190460-03ecc380-b20c-11e8-832b-839959ee2c99.gif)
 - [Significant head and base branches are added to the PR list.](https://user-images.githubusercontent.com/1402241/51428391-ae9ed500-1c35-11e9-8e54-6b6a424fede4.png)
 - [Hidden comments are previewed inline.](https://user-images.githubusercontent.com/1402241/52545036-6e271700-2def-11e9-8c0c-b5e0fa6f37dd.png)
+- [Find a user’s top repositories in their profile.](https://user-images.githubusercontent.com/13842856/52824495-da6f7800-3086-11e9-995d-ab8bd784e9ab.png)
```
Comment (bfred-it) [inline readme.md]: This was already in another section. It just needs a screenshot update (same style)

https://github.com/sindresorhus/refined-github#more-shortcuts

> [Find a user’s most starred repositories in their profile.](https://user-images.githubusercontent.com/1402241/48474026-43e3ae80-e82c-11e8-93de-159ad4c6f283.png)

**Comment by bfred-it** [comment]
Uhm, well this is awkward.

Isn't... the regular `Repositories` tab already showing the most-recently updated repositories?

screenshot 2019-02-20 at 21 16 36

The results seem to vary slightly because of what they consider an update... maybe? The tab marks CircularProgress as updated 3 hours ago and the search says 17... but:

screenshot 2019-02-20 at 21 19 19

screenshot 2019-02-20 at 21 19 52

In short I think the our new link isn't useful and it's also not entirely correct. 

---

Sorry I realized this 0.5s after I clicked Merge. 🤷‍♂️

**Comment by yakov116** [comment]
@bfred-it I was wondering if I missed something 😀 I thought the repository tab was auto sorted....
